### PR TITLE
feat: Add dark mode on web and android + match iOS style

### PR DESCRIPTION
**Pull request by Andresain123**
# Overview

This Pull Request adds support for the appearance prop on Web and Android.
I have also tweaked some of the styles to make it more close to the iOS one by adding shadows and the separators between the segments. Just like the iOS version if the tintColor or backgroundColor props are present, the separators dissappear.

An example of how it works:
<SegmentedControl appearance="dark" />

Fixes: #95  

# Test Plan

The example app is updated with this new functionality.

Some previews:
Screenshot 2020-09-21 at 01 14 20
Screenshot_20200921-011513

### CODE REVIEW

**Comment by exotexot** [comment]
Tested the pull request. See image attached. After the PR the label is missing and hence also not clickable.

segemented-darkmode-issue

**Comment by Andresain123** [comment]
@exotexot ok thanks, I didn't test it on react-native-windows. Let me check it and I'll try to solve it 👍

**Comment by exotexot** [comment]
But I can confirm that it is working on Android :)

Regarding the windows Issue. I'm aware of some styling constraints for the <Text> component in react-native-windows. There may be some conflicting styling properties which cause the windows issue.

File: js/SegmentedControl.ios.js [hunk 1]
```diff
@@ -99,7 +100,7 @@ class SegmentedControlIOS extends React.Component<Props> {
 
 const styles = StyleSheet.create({
   segmentedControl: {
-    height: 28,
+    height: 32,
```
Comment (Naturalclar) [inline js/SegmentedControl.ios.js:103]: just wanted to confirm, is the height change here intended?
This only changes the looks on iOS side, which seems out of scope for this PR
(if the change is intended, theres no problem :+1:)

Comment (Andresain123) [reply js/SegmentedControl.ios.js:103]: Yes, it was intended to match the exact height of the Segmented Control on iOS and the one for web/android.

**Review by Naturalclar — APPROVED** [review]
@Andresain123 thank you!

**Comment by exotexot** [comment]
Just wanted to add, that in my case I had to apply this change in order for darkmode to work properly in Windows:
https://github.com/exotexot/segmented-control/commit/e2c605b0a8d58095d97880647755e84c6fdf01ce

Setting the height appread to cause the problem, which I didnt investigate any further. This fix also closes https://github.com/react-native-segmented-control/segmented-control/issues/134
